src/track/text-style.js

```diff
@@ -138,17 +140,31 @@ class TextStyle {
     return 'rgba(' + color.concat(opacity).join(',') + ')';
   }
 
+  static get defaultValues(): PKTextStyleObject {
+    return {
+      fontEdge: TextStyle.EdgeStyles.NONE,
+      fontSize: '100%',
+      fontScale: 1,
+      fontColor: TextStyle.StandardColors.WHITE,
+      fontOpacity: TextStyle.StandardOpacities.OPAQUE,
+      backgroundColor: TextStyle.StandardColors.BLACK,
+      backgroundOpacity: TextStyle.StandardOpacities.OPAQUE,
+      fontFamily: TextStyle.FontFamily.SANS_SERIF
+    };
+  }
+
   static fromJson(setting: PKTextStyleObject): TextStyle {
-    let clonedTextStyle = new TextStyle();
-    clonedTextStyle.fontEdge = setting.fontEdge || clonedTextStyle.fontEdge;
-    clonedTextStyle.fontSize = setting.fontSize || clonedTextStyle.fontSize;
-    clonedTextStyle.fontScale = setting.fontScale || clonedTextStyle.fontScale;
-    clonedTextStyle.fontColor = setting.fontColor || clonedTextStyle.fontColor;
-    clonedTextStyle.fontOpacity = setting.fontOpacity || clonedTextStyle.fontOpacity;
-    clonedTextStyle.backgroundColor = setting.backgroundColor || clonedTextStyle.backgroundColor;
-    clonedTextStyle.backgroundOpacity = setting.backgroundOpacity || clonedTextStyle.backgroundOpacity;
-    clonedTextStyle.fontFamily = setting.fontFamily || clonedTextStyle.fontFamily;
-    return clonedTextStyle;
+    const currentSetting = Utils.Object.mergeDeep(TextStyle.defaultValues, setting);
+    let textStyle = new TextStyle();
+    textStyle.fontEdge = currentSetting.fontEdge;
+    textStyle.fontSize = currentSetting.fontSize;
+    textStyle.fontScale = currentSetting.fontScale;
+    textStyle.fontColor = currentSetting.fontColor;
+    textStyle.fontOpacity = currentSetting.fontOpacity;
+    textStyle.backgroundColor = currentSetting.backgroundColor;
+    textStyle.backgroundOpacity = currentSetting.backgroundOpacity;
+    textStyle.fontFamily = currentSetting.fontFamily;
+    return textStyle;
```

Comment:
In general I think the better approach here is to use a helper method to check if settings exist and not null or undefined:
```javascript
function get(value, default) {
  let result = default;
  if (value && value !== null && value !== undefined) {
    result = value;
  }
  return result;
}


let textStyle = new TextStyle();
textStyle.fontEdge = get(setting.fontEdge, clonedTextStyle.fontEdge);
textStyle.fontSize = get(setting.fontSize, clonedTextStyle.fontSize);
textStyle.fontScale = get(setting.fontScale, clonedTextStyle.fontScale);
textStyle.fontColor = get(setting.fontColor, clonedTextStyle.fontColor);
textStyle.fontOpacity = get(setting.fontOpacity, clonedTextStyle.fontOpacity);
textStyle.backgroundColor = get(setting.backgroundColor, clonedTextStyle.backgroundColor);
textStyle.backgroundOpacity = get(setting.backgroundOpacity, clonedTextStyle.backgroundOpacity);
textStyle.fontFamily = get(setting.fontFamily, clonedTextStyle.fontFamily;
return textStyle;

```

and if we take it further, there's a new JS feature called [nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) that does exactly this:

```javascript
textStyle.fontEdge = setting.fontEdge ?? clonedTextStyle.fontEdge;
```

and there's a babel plugin for it: @babel/plugin-proposal-nullish-coalescing-operator

Reply:
this one looks better to you?? I thought it will be better to merge cause it will take the value if exists

Reply:
If all of the ECMA script experts that write the JavaScript spec think this is good and this is a supported feature already in browsers(check chrome dev tools now if you want 😀 ) then it is good enough to me.

Reply:
I mean the get helper :)